src/clientsettingsdlg.cpp

```diff
@@ -804,3 +813,31 @@ void CClientSettingsDlg::OnInputBoostChanged()
     pSettings->iInputBoost = cbxInputBoost->currentIndex() + 1;
     pClient->SetInputBoost ( pSettings->iInputBoost );
 }
+
+void CClientSettingsDlg::contextMenuRequest()
+{
+    // add line to context menu, is dynamic, made at each request
+    QMenu* menu = cbxCentralServerAddress->lineEdit()->createStandardContextMenu();
```

Comment:
This seems like unnecessary overhead.  Define the menu once, on start up, with the extra item in it, all connected up.  Then here, just call it's popup action.

Reply:
Maybe keep the "connect it up" separate, so the client and server can share the menu.